This fixes problems with DVI monitors connected to the HDMI port
via a DVI <-> HDMI cable. With dvi monitors, the list of CEA modes
is always zero, preventing modes higher than 1024x768 to be used.
This patch disables the CEA mode check.

Upstream-Status: Pending

diff --git a/drivers/video/mxc_hdmi.c b/drivers/video/mxc_hdmi.c
index 544f352..fa67128 100644
--- a/drivers/video/mxc_hdmi.c
+++ b/drivers/video/mxc_hdmi.c
@@ -1804,10 +1804,10 @@ static void mxc_hdmi_edid_rebuild_modelist(struct mxc_hdmi *hdmi)
 		 */
 		mode = &hdmi->fbi->monspecs.modedb[i];
 
-		if (!(mode->vmode & FB_VMODE_INTERLACED) &&
-				(mxc_edid_mode_to_vic(mode) != 0)) {
+		if (!(mode->vmode & FB_VMODE_INTERLACED)) {
+			int vic = mxc_edid_mode_to_vic(mode);
 
-			dev_dbg(&hdmi->pdev->dev, "Added mode %d:", i);
+			dev_dbg(&hdmi->pdev->dev, "%s: Added mode %d(VIC %u):", __func__, i, vic);
 			dev_dbg(&hdmi->pdev->dev,
 				"xres = %d, yres = %d, freq = %d, vmode = %d, flag = %d\n",
 				hdmi->fbi->monspecs.modedb[i].xres,
